py-polars/tests/unit/test_df.py

```diff
@@ -1319,6 +1319,27 @@ def test_from_rows() -> None:
     assert data == df.rows()
 
 
+def test_from_rows_of_dicts() -> None:
+    records = [
+        {"id": 1, "value": 100, "_meta": "a"},
+        {"id": 2, "value": 101, "_meta": "b"},
+    ]
+    df1 = pl.DataFrame(records)
+    assert df1.rows() == [(1, 100, "a"), (2, 101, "b")]
+
+    overrides = {
+        "id": pl.Int16,
+        "value": pl.Int32,
+    }
+    df2 = pl.DataFrame(records, schema_overrides=overrides)
+    assert df2.rows() == [(1, 100, "a"), (2, 101, "b")]
+    assert df2.schema == {"id": pl.Int16, "value": pl.Int32, "_meta": pl.Utf8}
+
+    df3 = pl.DataFrame(records, schema=overrides)
+    assert df3.rows() == [(1, 100), (2, 101)]
+    assert df3.schema == {"id": pl.Int16, "value": pl.Int32}
```

Comment:
Wait, is this allowed? I thought when specifying `schema`, the input must always match the number of columns?

Reply:
I think I need to update the docstring! For lists of dicts this is both allowed and exactly how to do it; will update this PR with a better explanation and example in the docstring. (I'll also quickly check the other methods' do strings to see if they are accurate!)

Reply:
Done: https://github.com/pola-rs/polars/pull/6431